requirements.txt

```diff
@@ -29,4 +29,8 @@ pycocotools>=2.0.2
 py-cpuinfo>=8.0.0
 shapely>=2.0.0
 fire==0.5.0
+optuna==3.2.0
+plotly==5.16.1
 Pillow==9.5.0
+kaleido==0.2.1
+pyarrow==11.0.0
```

Comment:
pyarrow is needed because of the problem of FinalizeS3 was not called even though S3 was initialized .
plotly kaleido is needed to visualize and save optuna plots